crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -397,6 +464,13 @@ impl AcirContext {
     /// either the key or the value.
     fn add_data(&mut self, data: AcirVarData) -> AcirVar {
         assert_eq!(self.data.len(), self.data_reverse_map.len());
+        // if let Some(acir_var) = self.data_reverse_map.get(&data) {
```

Comment:
Seems like a bit of tech debt here

Reply:
Removed. With the 1-bit optimised XOR removed I'm no longer hitting this. Either way, whether or this change makes it into the codebase is in the hands of https://github.com/noir-lang/noir/pull/1470